SteamKit2/SteamKit2/Steam/Handlers/SteamUnifiedMessages/Callbacks.cs

```diff
@@ -3,114 +3,33 @@
  * file 'license.txt', which is part of this source code package.
  */
 
-using System;
-using System.Reflection;
 using ProtoBuf;
 
 namespace SteamKit2
 {
-    public partial class SteamUnifiedMessages : ClientMsgHandler
+    public partial class SteamUnifiedMessages
     {
         /// <summary>
         /// This callback is returned in response to a service method sent through <see cref="SteamUnifiedMessages"/>.
         /// </summary>
-        public class ServiceMethodResponse : CallbackMsg
+        public class ServiceMsg<TResult> : CallbackMsg where TResult : IExtensible, new()
         {
             /// <summary>
             /// Gets the result of the message.
             /// </summary>
             public EResult Result { get; private set; }
 
             /// <summary>
-            /// Gets the name of the Service.
+            /// Gets the protobuf body.
             /// </summary>
-            public string ServiceName
-            {
-                get { return MethodName.Split( '.' )[ 0 ]; }
-            }
-
-            /// <summary>
-            /// Gets the name of the RPC method.
-            /// </summary>
-            public string RpcName
-            {
-                get { return MethodName[ ( ServiceName.Length + 1 ).. ].Split( '#' )[ 0 ]; }
-            }
-
-            /// <summary>
-            /// Gets the full name of the service method.
-            /// </summary>
-            public string MethodName { get; private set; }
-
-            private PacketClientMsgProtobuf PacketMsg;
+            public TResult MessageBody { get; private set; }
```

Comment:
```suggestion
            public TResult Body { get; private set; }
```

we use `Body` in `ClientMsgProtobuf<TBody>`